Fix conflicting types true/false

### Problem Description

Compiling on some platforms can result in this problem because true/false are already defined elsewhere (and C23 promotes them to keywords now):

```
pkcs11-tool.c: In function 'derive_ec_key':
pkcs11-tool.c:4690:18: error: expected identifier or '(' before numeric constant
 4690 |         CK_BBOOL true = TRUE;
      |                  ^~~~
pkcs11-tool.c:4691:18: error: expected identifier or '(' before numeric constant
 4691 |         CK_BBOOL false = FALSE;
      |                  ^~~~~
pkcs11-tool.c:4694:29: error: lvalue required as unary '&' operand
 4694 |                 {CKA_TOKEN, &false, sizeof(false)}, /* session only object */
      |                             ^
pkcs11-tool.c:4697:33: error: lvalue required as unary '&' operand
 4697 |                 {CKA_SENSITIVE, &false, sizeof(false)},
      |                                 ^
pkcs11-tool.c:4698:35: error: lvalue required as unary '&' operand
 4698 |                 {CKA_EXTRACTABLE, &true, sizeof(true)},
      |                                   ^
pkcs11-tool.c:4699:31: error: lvalue required as unary '&' operand
 4699 |                 {CKA_ENCRYPT, &true, sizeof(true)},
      |                               ^
pkcs11-tool.c:4700:31: error: lvalue required as unary '&' operand
 4700 |                 {CKA_DECRYPT, &true, sizeof(true)},
      |                               ^
pkcs11-tool.c:4701:28: error: lvalue required as unary '&' operand
 4701 |                 {CKA_WRAP, &true, sizeof(true)},
      |                            ^
pkcs11-tool.c:4702:30: error: lvalue required as unary '&' operand
 4702 |                 {CKA_UNWRAP, &true, sizeof(true)}
      |                              ^
```

### Proposed Resolution

Just do as the rest of the code and use _true/_false instead

### Steps to reproduce

### Logs


Will post a PR to fix this.

--- a/src/tests/p11test/p11test_case_ec_derive.c
+++ b/src/tests/p11test/p11test_case_ec_derive.c
@@ -32,19 +32,19 @@ pkcs11_derive(test_cert_t *o, token_info_t * info,
 	CK_OBJECT_CLASS newkey_class = CKO_SECRET_KEY;
 	CK_KEY_TYPE newkey_type = CKK_GENERIC_SECRET;
 	CK_ULONG newkey_len = o->bits / 8;
-	CK_BBOOL true = TRUE;
-	CK_BBOOL false = FALSE;
+	CK_BBOOL _true = TRUE;
+	CK_BBOOL _false = FALSE;
 	CK_ATTRIBUTE template[] = {
-		{CKA_TOKEN, &false, sizeof(false)}, /* session only object */
+		{CKA_TOKEN, &_false, sizeof(_false)}, /* session only object */
 		{CKA_CLASS, &newkey_class, sizeof(newkey_class)},
 		{CKA_KEY_TYPE, &newkey_type, sizeof(newkey_type)},
 		{CKA_VALUE_LEN, &newkey_len, sizeof(newkey_len)},
-		{CKA_SENSITIVE, &false, sizeof(false)},
-		{CKA_EXTRACTABLE, &true, sizeof(true)},
-		{CKA_ENCRYPT, &true, sizeof(true)},
-		{CKA_DECRYPT, &true, sizeof(true)},
-		{CKA_WRAP, &true, sizeof(true)},
-		{CKA_UNWRAP, &true, sizeof(true)}
+		{CKA_SENSITIVE, &_false, sizeof(_false)},
+		{CKA_EXTRACTABLE, &_true, sizeof(_true)},
+		{CKA_ENCRYPT, &_true, sizeof(_true)},
+		{CKA_DECRYPT, &_true, sizeof(_true)},
+		{CKA_WRAP, &_true, sizeof(_true)},
+		{CKA_UNWRAP, &_true, sizeof(_true)}
 	};
 	CK_ATTRIBUTE get_value = {CKA_VALUE, NULL_PTR, 0};
 	CK_ULONG template_len = 10;
@@ -287,7 +287,7 @@ int test_derive(test_cert_t *o, token_info_t *info, test_mech_t *mech)
 		EVP_PKEY_free(evp_pkey);
 		return 1;
 	}
-	
+
 	/* Get buffer length */
 	if (EVP_PKEY_derive(pctx, NULL, &secret_len) != 1) {
 		debug_print(" [ KEY %s ] EVP_PKEY_derive failed", o->id_str);
@@ -302,7 +302,7 @@ int test_derive(test_cert_t *o, token_info_t *info, test_mech_t *mech)
 		EVP_PKEY_free(evp_pkey);
 		return 1;
 	}
-	
+
 	if (EVP_PKEY_derive(pctx, secret, &secret_len) != 1) {
 		debug_print(" [ KEY %s ] EVP_PKEY_derive failed", o->id_str);
 		EVP_PKEY_CTX_free(pctx);
@@ -311,7 +311,7 @@ int test_derive(test_cert_t *o, token_info_t *info, test_mech_t *mech)
 		return 1;
 	}
 	EVP_PKEY_CTX_free(pctx);
-	
+
 	/* Try to do the same with the card key */
 
 	/* Get length of pub */

--- a/src/tools/pkcs11-tool.c
+++ b/src/tools/pkcs11-tool.c
@@ -4765,19 +4765,19 @@ derive_ec_key(CK_SESSION_HANDLE session, CK_OBJECT_HANDLE key, CK_MECHANISM_TYPE
 	CK_MECHANISM mech;
 	CK_OBJECT_CLASS newkey_class= CKO_SECRET_KEY;
 	CK_KEY_TYPE newkey_type = CKK_GENERIC_SECRET;
-	CK_BBOOL true = TRUE;
-	CK_BBOOL false = FALSE;
+	CK_BBOOL _true = TRUE;
+	CK_BBOOL _false = FALSE;
 	CK_OBJECT_HANDLE newkey = 0;
 	CK_ATTRIBUTE newkey_template[20] = {
-		{CKA_TOKEN, &false, sizeof(false)}, /* session only object */
+		{CKA_TOKEN, &_false, sizeof(_false)}, /* session only object */
 		{CKA_CLASS, &newkey_class, sizeof(newkey_class)},
 		{CKA_KEY_TYPE, &newkey_type, sizeof(newkey_type)},
-		{CKA_SENSITIVE, &false, sizeof(false)},
-		{CKA_EXTRACTABLE, &true, sizeof(true)},
-		{CKA_ENCRYPT, &true, sizeof(true)},
-		{CKA_DECRYPT, &true, sizeof(true)},
-		{CKA_WRAP, &true, sizeof(true)},
-		{CKA_UNWRAP, &true, sizeof(true)}
+		{CKA_SENSITIVE, &_false, sizeof(_false)},
+		{CKA_EXTRACTABLE, &_true, sizeof(_true)},
+		{CKA_ENCRYPT, &_true, sizeof(_true)},
+		{CKA_DECRYPT, &_true, sizeof(_true)},
+		{CKA_WRAP, &_true, sizeof(_true)},
+		{CKA_UNWRAP, &_true, sizeof(_true)}
 	};
 	int n_attrs = 9;
 	CK_ECDH1_DERIVE_PARAMS ecdh_parms;

